site/components/pages/Retirements/SingleRetirement/ViewPledgeButton/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { ButtonPrimary } from "@klimadao/lib/components";
+import { t } from "@lingui/macro";
+import { Pledge } from "components/pages/Pledge/types";
+import { FC } from "react";
+
+type ViewPledgeButtonProps = {
+  pledge: Pledge | null;
+};
+
+/** Logic for hiding or showing a link to the users pledge depending on a valid pledge */
+export const ViewPledgeButton: FC<ViewPledgeButtonProps> = (props) => {
```

Comment:
Nitpicking, no action necessary:

I feel like we can remove the ternary operators in this component and be explicit with what we return based on whether there is a pledge present.

```js
export const ViewPledgeButton: FC<ViewPledgeButtonProps> = (props) => {
  if (props.pledge) {
    <ButtonPrimary
      href={`https://www.klimadao.finance/pledge/${props.pledge.ownerAddress}`}
      target="_blank"
      rel="noopener noreferrer"
      label={t({
        id: "retirement.single.view_pledge",
        message: "View Pledge",
      })}
    />;
  }

  return (
    <ButtonPrimary
      href={undefined}
      disabled={true}
      label={t({
        id: "retirement.single.no_pledge",
        message: "This user has not created a pledge",
      })}
    />
  );
};
```

This reads a lot more clearly to me. What do you think?

Reply:
I agree, much cleaner. Updated now.